ui/src/widgets/ui-markdown/UIMarkdown.vue

```diff
@@ -45,12 +71,31 @@ export default {
                 ...mapState('data', ['messages']),
                 ...mapGetters('data', ['getMsgProperty']),
                 msg () {
+                    if (this.messages[this.id]) {
+                        // re-render chart with new input
+                        this.renderMermaid()
+                    }
                     return this.messages[this.id] || {}
                 }
             },
+            mounted () {
+                this.renderMermaid()
+            },
             methods: {
                 get (path, defaultValue) {
                     return this.getMsgProperty(this.id, path, defaultValue)
+                },
+                renderMermaid () {
+                    // remove hte flag that mermaid uses to work out if an element has been processed
+                    // TODO: need to scope this to _just_ this component, otherwise it'll run for all mermaid charts on the page
+                    document.getElementsByClassName('mermaid')[0]?.removeAttribute('data-processed')
+                    this.$nextTick(() => {
+                        // let Vue render the dynamic markdown first, then re-render the chart
+                        mermaid.run({
+                            querySelector: '.mermaid',
+                            suppressErrors: true
+                        })
+                    })
```

Comment:
Rather than setting some scoped class, we can be specific about which nodes in the DOM to mermaidize.

e.g. instead of setting `querySelector: '.mermaid'` we can set `nodes: document.querySelectorAll('THIS_WIDGET_ID .mermaid')` 


See https://mermaid.js.org/config/usage.html#using-mermaid-run

Reply:
oh, wait, you mean be more specific on this `document.getElementsByClassName('mermaid')[0]?.removeAttribute('data-processed')` 

Yes, I agree.